frontend/src/components/ExpensePage.js

```diff
@@ -1,21 +1,50 @@
-import React from "react";
+import React, { Component } from "react";
 import Header from "./Header";
 import ExpensePresentation from "./ExpensePresentation";
 import ExpenseList from "./ExpenseList";
 import Footer from "./Footer";
+
 import "../styles/ExpensePage.css";
 
-function ExpensePage() {
-  return (
-    <div className="ExpensePage">
-      <Header />
-      <main>
-        <ExpensePresentation />
-        <ExpenseList />
-      </main>
-      <Footer />
-    </div>
-  );
+export class ExpensePage extends Component {
+  state = {
+    total: 0,
+  };
+
+  updateTotal = (amount) => {
+    this.setState({ total: this.state.total + parseFloat(amount) });
+  };
+
+  // React Life Cycle
+  componentDidMount() {
+    this.total = JSON.parse(localStorage.getItem("total"));
+    if (localStorage.getItem("total")) {
+      this.setState({
+        total: this.total,
+      });
+    } else {
+      this.setState({
+        total: 0,
+      });
+    }
+  }
+
+  componentWillUpdate(nextProps, nextState) {
+    localStorage.setItem("total", JSON.stringify(nextState.total));
+  }
```

Comment:
componentWillUpdate is apparently an unsafe legacy method, we should rather use e.g. componentDidUpdate and read the total from the new state directly:
``` js
 componentDidUpdate() {
    localStorage.setItem("total", JSON.stringify(this.state.total));
  }
```

Reply:
Worked like a charm, thanks for the tip!